numpy/polynomial/polynomial.py

```diff
@@ -776,7 +776,11 @@ def polyval(x, c, tensor=True):
 
     c0 = c[-1] + x*0
     for i in range(2, len(c) + 1):
-        c0 = c[-i] + c0*x
+        # separate out the Horner operation c0 = c[-i] + c0*x,
```

Comment:
Looks like we are OK here. I note that there should be a check that if `isinstance(x, ndarray) and not tensor`, then `x.ndim < c.ndim` is true. The same check is probably missing in all the other polynomial types, but that is not something for this PR.